### DISCUSSION TITLE
Where do I use --initial-prompt in python code?

### DISCUSSION THREAD

**Discussion by GauravCodesAI** [openai/whisper#355] [Q&A] [answered]
How do I suggest vocabulary in code?
Thank You,

**Comment by jianfch** [comment, marked as answer]
```python
model = whisper.load_model("base")
result = model.transcribe("audio.mp3", initial_prompt='vocabulary' )
```

**Reply by GauravCodesAI** [reply]
Thank You!, Anyway to do it using,
```
options = whisper.DecodingOptions(fp16=False)
result = whisper.decode(model, mel, options)
```

**Reply by GauravCodesAI** [reply]
NVM Got it

`options = whisper.DecodingOptions(fp16=False, prompt="vocab")`

**Reply by madcap1090** [reply]
Hi there. I thought it was a list of words, but in fact it's a prompt, like the other models. See https://platform.openai.com/docs/guides/speech-to-text/prompting

**Reply by envious** [reply]
Does anyone know how to achieve this with the official OpenAI Whisper API? i.e. add prompts

**Reply by aavetis** [reply]
> Does anyone know how to achieve this with the official OpenAI Whisper API? i.e. add prompts

`prompt` in [params listed here](https://platform.openai.com/docs/api-reference/audio/create)

**Comment by sthita-pujari** [comment]
@DracoCoder Just to understand this? Is 'vocab' a list of words?
What I wanted is a list of keywords that I can use to boost the LM while decoding.
Is there a way to do that?

**Reply by GauravCodesAI** [reply]
>Just to understand this? Is 'vocab' a list of words?

Yes, It's a string of words expected to be present in the audio and should boost speech recognition.

**Reply by sthita-pujari** [reply]
So, if you want to add "Tensorflow" and "Pytorch" to boost.
My input would be `prompt="Tensorflow pytorch"` ?

**Reply by GauravCodesAI** [reply]
Yes

**Reply by wqw547243068** [reply]
> Yes

Any restriction on prompt content? such as length, and punctuation?

**Reply by esonec** [reply]
Can my vocabulary contain 2-word phrases? Can it contain '-'-separated words?
Like "orthogonal UUID" or moisture-less.
Then how should I add them to the initial_prompt amongst simple one word terms?

**Reply by envious** [reply]
> Can my vocabulary contain 2-word phrases? Can it contain '-'-separated words? Like "orthogonal UUID" or moisture-less. Then how should I add them to the initial_prompt amongst simple one word terms?

I'm keen to know this too

**Reply by zaherweb** [reply]
I think it can contain different words, and actually model works with them separately I have tested that , but the problem is that you are limited to total lenght of 1024 , as prompt deals with the whole text as a string not a list.

**Reply by glangford** [reply]
> I think it can contain different words, and actually model works with them separately I have tested that , but the problem is that you are limited to total lenght of 1024 , as prompt deals with the whole text as a string not a list.

@zaherweb Why would you have a limit of 1024? Python strings can be much longer than 1024, is there some other limitation you are running into?

**Reply by glangford** [reply]
...probably you are running from the command line and running into a shell limitation?

**Comment by KyleFRS** [comment]
@DracoCoder could you please show the full file for the script? I also would like to know but I couldn't follow the answer. Thank you.
